File: application/src/main/java/com/example/server/controller/UserController.java
package com.example.server.controller;

import com.example.api.UserApi;
import com.example.api.dto.UserDto;
import com.example.server.service.UserService;
import com.example.server.mapper.UserMapper;
import com.example.server.repository.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping(path = UserApi.PATH)
@RequiredArgsConstructor
public class UserController implements UserApi {

    private final UserMapper userMapper;
    private final UserService userService;

    @Override
    public UserDto create(UserDto userDto) {
        User user = userMapper.toUser(userDto);

        return userMapper.toUserDto(userService.create(user));
    }

    @Override
    public UserDto update(Long id, UserDto userDto) {
        User user = userMapper.updateUser(userDto, userService.findById(id));

        return userMapper.toUserDto(userService.update(user));
    }

    @Override
    public UserDto findById(Long id) {
        return userMapper.toUserDto(userService.findById(id));
    }

    @Override
    public List<UserDto> findAll() {
        return userMapper.toUserDto(userService.findAll());
    }

    @Override
    public Map<String, String> deleteById(Long id) {
        userService.deleteById(id);
        return Map.of("message", "Пользователь с указанным идентификатором был успешно удален.");
    }
}

File: application/src/main/kotlin/com/example/server/controller/UserController.kt
package com.example.server.controller

import com.example.api.UserApi
import com.example.api.model.UserDto
import com.example.server.mapper.toDto
import com.example.server.mapper.toEntity
import com.example.server.service.UserService
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [UserApi.PATH])
class UserController(
    private val userService: UserService
) : UserApi {

    override fun create(userDto: UserDto): UserDto {
        val user = userDto.toEntity()

        return userService.create(user).toDto()
    }

    override fun update(id: Long, userDto: UserDto): UserDto {
        return userService.update(id, userDto).toDto()
    }

    override fun findById(id: Long): UserDto {
        return userService.findById(id).toDto()
    }

    override fun findAll(): List<UserDto> {
        return userService.findAll().toDto()
    }

    override fun deleteById(id: Long) {
        userService.deleteById(id)
    }
}
